shared/validate/validate.go

```diff
@@ -542,12 +538,8 @@ func IsUUID(value string) error {
 
 // IsPCIAddress validates whether a value is a PCI address.
 func IsPCIAddress(value string) error {
-	regexHex, err := regexp.Compile(`^([0-9a-fA-F]{4}?:)?[0-9a-fA-F]{2}:[0-9a-fA-F]{2}\.[0-9a-fA-F]$`)
-	if err != nil {
-		return err
-	}
-
-	if !regexHex.MatchString(value) {
+	match, _ := regexp.MatchString(`^(?:[0-9a-fA-F]{4}:)?[0-9a-fA-F]{2}:[0-9a-fA-F]{2}\.[0-9a-fA-F]$`, value)
```

Comment:
@simondeziel what does this change do?

Can you confirm you've tested this, as not all of the usual regex functionality (i.e those in perl) are available to Go. Thanks

Reply:
It does 2 things:

* Adding `?:` at the beginning of group `()`, tells the engine to not capture/remember the matched string
* Replace `{4}?` by the more common `{4}` which says to match for exactly `4` repetitions

See the "Grouping" and "Repetitions" section on https://github.com/google/re2/wiki/Syntax which is essentially the syntax accepted by https://pkg.go.dev/regexp.

Yes, I tested all the regexp changes with the playground, here's the `IsPCIAddress()` one https://play.golang.com/p/L_1hpVSNEao

Reply:
Thanks!